Heatington/Controllers/CsvController.cs

```diff
@@ -23,6 +25,19 @@ public class CsvController : IDataSource
         }
 
         // TODO: Check if this method is actually required
-        public void SaveData(List<DataPoint> data, string filePath) => throw new NotImplementedException();
+        public void SaveData(List<DataPoint> data)
+        {
+            try
+            {
+                CsvData csv = CsvData.Create(data);
```

Comment:
I am confused, I don't even understand how dotnet allows this. Shouldn't this be:
```csharp
CsvData csv = CsvData.Create<DataPoint>(data);
```
You haven't changed the CsvData class as far as I can see and the functoin definition there is Create<T>, so I don't know how you are using without a T. Unless C# is smart and it can figure it out that since data is a List of DataPoints and the parameter is List<T> then T must be DataPoints. But even if that's true I would be explicit to avoid confusion.

Reply:
Yeah C# already knows that you want to cast it to different type, but yeah explicitly defining would be more readable

Reply:
@0rangeBlaze fixed!